impeller/renderer/command.h

```diff
@@ -60,13 +60,12 @@ struct Resource {
 
 using BufferResource = Resource<BufferView>;
 using TextureResource = Resource<std::shared_ptr<const Texture>>;
-using SamplerResource = Resource<std::shared_ptr<const Sampler>>;
 
 /// @brief combines the texture, sampler and sampler slot information.
 struct TextureAndSampler {
   SampledImageSlot slot;
   TextureResource texture;
-  SamplerResource sampler;
+  std::shared_ptr<const Sampler> sampler;
```

Comment:
We don't need a Resource wrapper for the sampler since we only support combined image+sampler.